### PR TITLE
Spark sql to hdfs

### PR DESCRIPTION
**Pull request by shikham-8**
- removed POM dependencies
- Spark SQL exporter

### CODE REVIEW

**Review by jcgubbi — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 build:
 	mvn clean package
 
-run:
+run: build
```
Comment (brendon-ng) [inline Makefile]: this should be left as just `run: `

File: pom.xml [hunk 2]
```diff
@@ -29,27 +29,7 @@
...
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-tools</artifactId>
-      <version>1.2.1</version>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-client</artifactId>
-      <version>2.6.0</version>
+
```
Comment (brendon-ng) [inline pom.xml]: Lets see if the HDFS stuff can run without all this

File: pom.xml [hunk 3]
```diff
@@ -126,25 +106,25 @@
...
+                </filter>
+              </filters>
+            </configuration>
           </execution>
         </executions>
       </plugin>
     </plugins>
   </build>
-</project>
+</project>
```
Comment (brendon-ng) [inline pom.xml:129]: add newline at end of file

File: src/main/java/app/App.java [hunk 4]
```diff
@@ -7,9 +7,6 @@
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
 
-import app.SparkSQL.*;
-import app.hdfs.*;
-
```
Comment (brendon-ng) [inline src/main/java/app/App.java]: Let's not take these out if we don't have to pt 2.

File: src/main/java/app/App.java [hunk 5]
```diff
@@ -39,9 +36,6 @@ private static void run(String[] args) throws IOException {
                 case "SPARKSQL":
                     importer = new SparkSQLImporter(importerConfig);
                     break;
-                case "HDFS":
-                    importer = new HDFSImporter(importerConfig);
-                    break;
```
Comment (brendon-ng) [inline src/main/java/app/App.java]: Lets not take these out if we don't have to

File: src/main/java/app/SparkSQL/SparkSQLImporter.java [hunk 6]
```diff
@@ -24,13 +32,26 @@ protected void parseJSON(JSONObject config) {
...
+        // File[] csvFile = csvDirectory.listFiles(regex);
+        // System.out.println("FINDING FILE");
+        // System.out.println(csvFile);
+        try {
+            // HDFSUtils.copyFromLocal(csvFile[0].getName(), Constants.WORKING_DIR, "addr",
+            // 8080);
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
```
Comment (brendon-ng) [inline src/main/java/app/SparkSQL/SparkSQLImporter.java]: jw what does this stuff do

File: src/main/java/app/SparkSQLImporter.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        Dataset<Row> df = spark.read().option("header", "true").csv(inputFilePath);
+        df.createOrReplaceTempView("XTLInput");
+
+        // run initial query on file
+        Dataset<Row> sqlQuery = spark.sql(query);
+
+        // save results in CSV directory - use *.csv to get file with results
+        sqlQuery.write().csv("data/sparksqlExtracted.csv");
+    }
+}
```
Comment (brendon-ng) [inline src/main/java/app/SparkSQLImporter.java]: Why are there 2 SparkSQLImporter?

**Review by brendon-ng — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (shikham-8) [reply pom.xml]: it cannot :(

**Review by brendon-ng — APPROVED** [review]

**Review by brendon-ng — APPROVED** [review]

**Pull request merged**